catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -0,0 +1,223 @@
+      return Stream.empty();
+    }
+
+    String payload = getFileContent(file);
+    if (payload == null) {
+      LOGGER.debug("Problem reading {}", file.getName());
+      return Stream.empty();
+    }
+
+    try {
+      return Stream.of(Boon.fromJson(payload))
+          .map(List.class::cast)
+          .flatMap(List::stream)
+          .filter(Map.class::isInstance)
+          .map(Map.class::cast)
+          .map(mapper)
+          .filter(Objects::nonNull)
+          // Invoke a terminal operation so the exception can be handled here
+          .collect(Collectors.toList())
+          .stream();
+    } catch (ClassCastException e) {
+      LOGGER.warn(
+          "Could not load forms configuration in {}, JSON should be a list of maps",
+          file.getName());
+      return Stream.empty();
+    }
+  }
+
+  /** Parse the JSON map for initializing system form templates. */
+  @Nullable
+  private Metacard formMapper(Map map) {
+    LOGGER.debug("Starting form processing...");
+
+    String title = safeGet(map, "title", String.class);
+    String description = safeGet(map, "description", String.class);
+    String filterTemplateFile = safeGet(map, "filterTemplateFile", String.class);
+
+    if (anyNull(title, description, filterTemplateFile)) {
+      LOGGER.debug("Invalid entry in forms.json");
+      return null;
+    }
+
+    File xmlFile = configDirectory.toPath().resolve(filterTemplateFile).toFile();
+    if (!xmlFile.exists()) {
+      LOGGER.debug("Filter XML file does not exist: {}", filterTemplateFile);
+      return null;
+    }
+
+    String filterXml = getFileContent(xmlFile);
+    if (filterXml == null) {
+      LOGGER.debug("Error while reading filter XML file: {}", filterTemplateFile);
+      return null;
+    }
+
+    QueryTemplateMetacardImpl metacard = new QueryTemplateMetacardImpl(title, description);
+    metacard.setFormsFilter(filterXml);
+    return metacard;
+  }
+
+  /** Parse the JSON map for initializing system result templates. */
+  @Nullable
+  private Metacard resultsMapper(Map map) {
+    LOGGER.debug("Starting result processing...");
+
+    String title = safeGet(map, "title", String.class);
+    String description = safeGet(map, "description", String.class);
+    List<String> descriptors = safeGetList(map, "descriptors", String.class);
+
+    if (anyNull(title, description, descriptors)) {
+      LOGGER.debug("Invalid entry in results.json");
+      return null;
+    }
+
+    ResultTemplateMetacardImpl metacard = new ResultTemplateMetacardImpl(title, description);
+    metacard.setResultDescriptors(new HashSet<>(descriptors));
+    return metacard;
+  }
+
+  @Nullable
+  private static String getFileContent(File file) {
+    try (InputStream is = new FileInputStream(file)) {
+      return IOUtils.toString(is, "UTF-8");
+    } catch (IOException e) {
+      LOGGER.error("Problem reading from {}, {}", file.getName(), e.getMessage());
+      LOGGER.debug("Problem reading from {}", file.getName(), e);
```

Comment:
Is it necessary to log at 2 different levels?

Reply:
Yes. Per our guidelines, we should not be logging stack traces at the higher levels.

Reply:
Well, let me rephrase that. They are listed as "acceptable" at the higher levels, but they "should" be at the debug levels. I'm actually inclined to disagree with how lax those docs are about the higher levels. Only reason I could see stack traces at `error` is for mission critical failures. 
https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels